### PR TITLE
Fb1386

### PR DESCRIPTION
**Pull request by spiresheep**

### CODE REVIEW

File: Applications/WebPortal/WebApp/library/source/pages/account_page/profile_page/text_field.tsx [hunk 1]
```diff
@@ -17,11 +17,13 @@ enum Effects {
 interface Properties {
 
   /** Indicates the input field can not be interacted with. */
-  disabled?: boolean;
+  readonly?: boolean;
 
   /** The value to display in the field. */
   value?: string;
 
+  placeholder?: string;
```
Comment (kranar) [inline Applications/WebPortal/WebApp/library/source/pages/account_page/profile_page/text_field.tsx]: Document all properties.

File: Applications/WebPortal/WebApp/library/source/pages/account_page/profile_page/text_field.tsx [hunk 2]
```diff
@@ -138,25 +141,31 @@ export class TextField extends React.Component<Properties, State> {
   private static STYLE = StyleSheet.create({
     box: {
       boxSizing: 'border-box' as 'border-box',
-      width: '100%',
+      //width: '100%',
```
Comment (kranar) [inline Applications/WebPortal/WebApp/library/source/pages/account_page/profile_page/text_field.tsx]: Remove comment.

File: Applications/WebPortal/WebApp/library/source/pages/account_page/profile_page/text_field.tsx [hunk 3]
```diff
@@ -138,25 +141,31 @@ export class TextField extends React.Component<Properties, State> {
...
+        border: '1px solid #FFFFFF'
+      },
+      '::placeholder': {
+        color: '#8C8C8C'
       }
     },
     hoveredBox: {
       boxSizing: 'border-box' as 'border-box',
-      width: '100%',
+      //width: '100%',
```
Comment (kranar) [inline Applications/WebPortal/WebApp/library/source/pages/account_page/profile_page/text_field.tsx]: Remove comment.

File: Applications/WebPortal/WebApp/library/source/pages/account_directory_page/create_account_page.tsx [hunk 4]
```diff
@@ -123,16 +121,15 @@ export class CreateAccountPage extends React.Component<Properties, State> {
                 <FormEntry name='Username'
                     displaySize={this.props.displaySize}>
                   <TextField
-                    value={''}
+                    value=''
```
Comment (kranar) [inline Applications/WebPortal/WebApp/library/source/pages/account_directory_page/create_account_page.tsx]: This needs to be fixed for all string literals.

File: Applications/WebPortal/WebApp/library/source/pages/account_directory_page/create_account_page.tsx [hunk 5]
```diff
@@ -123,16 +121,15 @@ export class CreateAccountPage extends React.Component<Properties, State> {
...
                     placeholder={'Username'}
                     displaySize={this.props.displaySize}/>
                 </FormEntry>
                  <Dali.Padding size={CreateAccountPage.SMALL_PADDING}/>
                  <FormEntry name='Role(s)'
                     displaySize={this.props.displaySize}>
                   <div style={CreateAccountPage.STYLE.rolesWrapper}>
-                    <RolesField
-                      roles={this.state.roles}/>
+                    <RolesField roles={this.state.roles}/>
```
Comment (kranar) [inline Applications/WebPortal/WebApp/library/source/pages/account_directory_page/create_account_page.tsx]: I can't edit the roles, the field should be editable.

**Comment by spiresheep** [comment]
That's for the next case, this case is just for laying it all out.

**Pull request merged**